### PR TITLE
Fix invited devices are always created in root entity

### PR DESCRIPTION
**Pull request by btry**
Automatically generate rules to "route" imported devices in the entity of  the invitation.

@rafaelje the tag in  the inventory must be similar to **invitation_4564546564587**.

I'm unsure how I described you the tag on telegram (underscore or dash). We need to check and use the same !

### CODE REVIEW

**Comment by btry** [comment]
@DIOHz0r I'm planning to move later the code interacting with Fusion Inventory in the new file inc/fusioninventory.class.php. Especially portions of enrollment for import.

**Comment by rafaelje** [comment]
I prefer to use "_" underscore

**Comment by rafaelje** [comment]
This is a real example of the new information tag:

```
    <ACCOUNTINFO>
      <KEYNAME>TAG</KEYNAME>
      <KEYVALUE>invitation_b3aa766d6b62163799b6dbd6086a8281ec9ce32cac90da89e5e9968699ac623e</KEYVALUE>
    </ACCOUNTINFO>
```

**Comment by rafaelje** [comment]
This is the PR related on the Android Agent:
https://github.com/flyve-mdm/android-mdm-agent/pull/405

**Comment by btry** [comment]
Perfect I'll try today

File: inc/fusioninventory.class.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+   /**
+    * Creates or updates an entity rule
+    *
+    * @param PluginFlyvemdmInvitation $invitation invitation to handle
+    */
+   public function addInvitationRule(PluginFlyvemdmInvitation $invitation) {
+      $entityId = $invitation->getField(Entity::getForeignKeyField());
+      $rule = $this->getRule($entityId);
+
+      $ruleCriteria = new RuleCriteria();
```
Comment (DIOHz0r) [inline inc/fusioninventory.class.php]: I have a question for this function

Don't you think it's better for performance to just compare if the rule exist and is active to then do a return?

If the rule is not active but exist, update it and if it not exist just add it

Comment (btry) [reply inc/fusioninventory.class.php]: A rule is combination of 3 different itemtypes:
- rule itself
- one or several criterias (with a boolean connector AND or OR)
- one or several actions

As I need to manupulate several itemtypes with relatios between them I choose to implement a class with this specific logic. The invitation class is already quite big.

I had 2 choices : 
- one rule per invitation with a single criteria
- one rule with a lot of criterias, one criteria per invitation

I choosed the 2nd, w th a boolean connector OR

**Pull request merged**